MegaApiClient/MegaApiClient.cs

```diff
@@ -574,7 +574,10 @@ public Stream Download(INode node, CancellationToken? cancellationToken = null)
       DownloadUrlRequest downloadRequest = new DownloadUrlRequest(node);
       DownloadUrlResponse downloadResponse = this.Request<DownloadUrlResponse>(downloadRequest);
 
-      Stream dataStream = new BufferedStream(this.webClient.GetRequestRaw(new Uri(downloadResponse.Url)));
+      Uri downloadUrl = new Uri(downloadResponse.Url);
+      Stream dataStream = new SeekableReadStream(this.webClient.GetLength(downloadUrl), (buffer, bufferOffset, offset, count)
+        => this.webClient.GetRequestRawWithRange(

```

Comment:
I prefer to hide this logic in the SeekableStream class to have something more readable.
Could be 
```csharp
Stream dataStream = new SeekableReadStream(downloadUrl, downloadResponse.Size, this.webClient);
```

And the following changes in SeekableStream.cs
```csharp
private readonly Uri downloadUri;
private readonly IWebClient webClient;

public SeekableReadStream(Uri downloadUri, long length, IWebClient webClient)
{
  this.Length = length;
  this.downloadUri = downloadUri;
  this.webClient = webClient;
}
[...]
public override int Read(byte[] buffer, int offset, int count)
{
  [...]
  using (var stream = this.webClient.GetRequestRawWithRange(this.uri, Position, Position + count))
  {
    var readCount = stream.Read(buffer, offset, count);
    this.Position += readCount;
    return readCount;
  }
}
```